H_ecto_models.md

```diff
@@ -597,6 +597,73 @@ end
 ```
 Because we declared a relationship in `HelloPhoenix.User`, `%User{}` will now contain a videos property which starts out as an unloaded relationship. In order to properly display it in `render`, we'll need to tell Ecto to preload the field. Note that `Repo.preload/2` is smart enough to work on just one model or collection of them.
 
+### Working with Callback Hooks
+
+Continuing with our video sharing app example, suppose some of our content-uploading users are horrible trolls and one of their favorite pastimes is to initially upload a video of happy huskies playing with furry kittens, then, after gathering a large number of "likes", change the happy video to an extremely controversial and blatantly offensive one. Then, they would write an inflammatory article about how the users of our app "like" extremely offensive videos, and spread this article around on social media to damage our online reputation and maybe even destroy business.
+
+As developers, we'd like to keep our livelihoods and not become internet infamous / unemployable. In this case, we'd like to make our `HelloPhoenix.Video` reset its like "likes" counter when the video's owner updates the record. Fortunately, Ecto ships with model callbacks that we can use to help in this endeavor.
+
+In this case, we will instruct Ecto to use the `before_update` hook in `web/models/video.ex` like so:
+
+```elixir
+defmodule HelloPhoenix.Video do
+. . .
+  before_update :reset_likes_count
+  def reset_likes_count(changeset) do
+    changeset
+    |> Ecto.Changeset.put_change(:likes, 0)
+  end
+end
+```
+Now, when `Repo.update` is called for the `Video` model, the `likes` field will automatically be reset to zero. This should keep us safe from the ravages of an often hostile modern Internet!
+
+`Ecto.Model.Callbacks` actually ships with with a lot more than the `before_update` callback, in addition, it comes with:
+
+- `before_delete`: called before the adapter deletes our record from the database; if perhaps we have a foreign_key constraint from another table, we would use this hook to clean it up.
+- `after_delete`: called after the adapter deletes our record. If we wished to archive our record, we'd do it in this hook. 
+- `before_update`: called before the adapter updates our record. We used it above to reset the likes counter.
+- `after_update`: called after the adapter updates our record. If we wanted to update our user that his model has changed, now would do that here.
+- `before_insert`: called before the adapter first creates a new record. We would use it in cases when we wanted to infer default values to certain fields from the partial changeset we receive from the user.
+- `after_insert`: called after the adapter creates our record. We could use this to notify the user a new has been created under his account.
```

Comment:
What would you think about using a more gender-neutral term  for the possessive pronoun here? By that I mean use "their" instead of "his"?

Note to self: I should add this to the contributing guidelines.